.github/actions/use-pre-commit/action.yml

```diff
@@ -30,29 +30,23 @@ runs:
         key: pre-commit-${{ inputs.version }}-${{ hashFiles(inputs.config-file) }}
         path: |
           ~/.cache/pre-commit
-          ~/.pre-commit-venv
-
-    - name: Create virtualenv
-      if: steps.cache-pre-commit.outputs.cache-hit != 'true'
-      run: python -m venv ~/.pre-commit-venv
-      shell: bash
 
     - name: Install pre-commit
       if: steps.cache-pre-commit.outputs.cache-hit != 'true'
       run: |
-        source ~/.pre-commit-venv/bin/activate
-        python -m pip install pre-commit==2.19.0
+        set -eux
+        curl --proto '=https' --tlsv1.2 -sSL https://github.com/pre-commit/pre-commit/releases/download/v${{ inputs.version }}/pre-commit-${{ inputs.version }}.pyz > ~/.cache/pre-commit/pre-commit.pyz
```

Comment:
```diff
-      run: |
-        set -eux
-        curl --proto '=https' --tlsv1.2 -sSL https://github.com/pre-commit/pre-commit/releases/download/v${{ inputs.version }}/pre-commit-${{ inputs.version }}.pyz > ~/.cache/pre-commit/pre-commit.pyz
+      run: curl --proto '=https' --tlsv1.2 -sSL https://github.com/pre-commit/pre-commit/releases/download/v${{ inputs.version }}/pre-commit-${{ inputs.version }}.pyz > ~/.cache/pre-commit/pre-commit.pyz
```

`Github Action` show the first line of `run` since you have 1 command here (excluding the `set ...`) you can put directly `curl` and skipping `set ...`

Reply:
I reverted my add of `set -eux` then

However I'm style thinking we should use at least `-x` which is really convenient to see which command has made which output (and while we are at doing `set -x` why not doing the good&explicit `set -eux` ? ^^)

Reply:
When doing multiple command in a `run` yes using `set -x` or `set -eux` could be useful.
But with one command, you already see it in the log of github action

Reply:
I agree, the "set -eux on multi line bash script, nothing on single line script" seems like the right thing to do ;-)